### PR TITLE
[ISSUE #2818] Dubbo part

### PR DESCRIPTION
**Pull request by dragon-zhang**
see https://github.com/apache/incubator-shenyu/issues/2818
this PR only compatible with old version of dubbo client. If approved, I will add more compatible PR.

### CODE REVIEW

**Comment by KevinClair** [comment]
What's the difference?

**Comment by dragon-zhang** [comment]
Well, when upgrading version from `2.2` to `2.4.2`, the client needs to replace `@SoulDubboClient with `@ShenyuDubboClient`, which obviously requires a lot of cost to promote hundreds of people to modify it.

**Comment by dragon-zhang** [comment]
> What's the difference?

Well, when upgrading version from `2.2` to `2.4.2`, the client needs to replace `@SoulDubboClient` with `@ShenyuDubboClient`, which obviously requires a lot of cost to promote hundreds of people to modify it.

File: shenyu-client/shenyu-client-dubbo/shenyu-client-alibaba-dubbo/src/main/java/org/apache/shenyu/client/alibaba/dubbo/AlibabaDubboServiceBeanListener.java [hunk 1]
```diff
@@ -100,19 +101,19 @@ private void handler(final ServiceBean<?> serviceBean) {
...
-                publisher.publishEvent(buildMetaDataDTO(serviceBean, shenyuDubboClient, method));
+            SoulDubboClient dubboClient = AnnotatedElementUtils.findMergedAnnotation(method, SoulDubboClient.class);
+            if (Objects.nonNull(dubboClient)) {
+                publisher.publishEvent(buildMetaDataDTO(serviceBean, dubboClient, method));
             }
         }
     }
 
-    private MetaDataRegisterDTO buildMetaDataDTO(final ServiceBean<?> serviceBean, final ShenyuDubboClient shenyuDubboClient, final Method method) {
+    private MetaDataRegisterDTO buildMetaDataDTO(final ServiceBean<?> serviceBean, final SoulDubboClient dubboClient, final Method method) {
```
Comment (yu199195) [inline shenyu-client/shenyu-client-dubbo/shenyu-client-alibaba-dubbo/src/main/java/org/apache/shenyu/client/alibaba/dubbo/AlibabaDubboServiceBeanListener.java:111]: why is sloudubbo client

**Review by yu199195 — CHANGES_REQUESTED** [review]
not merged

**Comment by yu199195** [comment]
@dragon-zhang  add my wechat， xixy199195

**Comment by KevinClair** [comment]
> not merged

agree.

File: shenyu-client/shenyu-client-dubbo/shenyu-client-alibaba-dubbo/src/main/java/org/apache/shenyu/client/alibaba/dubbo/AlibabaDubboServiceBeanListener.java (hunk 1 above)
Comment (dragon-zhang) [reply shenyu-client/shenyu-client-dubbo/shenyu-client-alibaba-dubbo/src/main/java/org/apache/shenyu/client/alibaba/dubbo/AlibabaDubboServiceBeanListener.java:111]: @ShenyuDubboClient is a derived annotation of @SoulDubboClient.
```java
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE, ElementType.METHOD})
@Inherited
@SoulDubboClient(path = "")
public @interface ShenyuDubboClient {

    @AliasFor(annotation = SoulDubboClient.class)
    String path();

    @AliasFor(annotation = SoulDubboClient.class)
    String ruleName() default "";

    @AliasFor(annotation = SoulDubboClient.class)
    String desc() default "";

    @AliasFor(annotation = SoulDubboClient.class)
    boolean enabled() default true;
}
```
see `org.springframework.core.annotation.AliasFor` and `org.springframework.core.annotation.AnnotatedElementUtils#findMergedAnnotation`.

Comment (yu199195) [reply shenyu-client/shenyu-client-dubbo/shenyu-client-alibaba-dubbo/src/main/java/org/apache/shenyu/client/alibaba/dubbo/AlibabaDubboServiceBeanListener.java:111]: Is it for compatibility? soul version, we are not going to maintain

Comment (dragon-zhang) [reply shenyu-client/shenyu-client-dubbo/shenyu-client-alibaba-dubbo/src/main/java/org/apache/shenyu/client/alibaba/dubbo/AlibabaDubboServiceBeanListener.java:111]: Yes, just for compatibility. So be it, close this PR ?

Comment (yu199195) [reply shenyu-client/shenyu-client-dubbo/shenyu-client-alibaba-dubbo/src/main/java/org/apache/shenyu/client/alibaba/dubbo/AlibabaDubboServiceBeanListener.java:111]: @dragon-zhang thank you for pr， you already add my wechat ：xixy199195?   ,you can close this PR~

**Pull request closed without merging**